### DISCUSSION TITLE
Basic Zenoh Discovery

### DISCUSSION THREAD

**Discussion by smnrgrs** [eclipse-zenoh/roadmap#151] [Q&A] [answered]
I'm testing some basic examples in zenoh-c. My network doesn't support Multicast for various reasons. 
So if I run z_sub & z_pub it doesnt work because of lack of multicast scouting. Instead I can do:
z_sub -l tcp/1.2.3.4:7447
z_pub -e tcp/1.2.3.4:7447
and this does work, with me specifying my endpoints.

Is there any way to make these examples work using a Zenoh router? Will I always need to specify endpoints without multicast working?

Thanks

**Comment by gabrik** [comment, marked as answer]
Hi @smnrgrs, you can use a Zenoh router as randevous-point and leverage on gossip discovery for the applications to connect directly, please refer to the reference manual section gossip: https://zenoh.io/docs/getting-started/deployment/#scouting

**Reply by smnrgrs** [reply]
Thanks
